patch/llpcPatchResourceCollect.cpp

```diff
@@ -2372,6 +2402,385 @@ void PatchResourceCollect::MapGsBuiltInOutput(
+    {
+        m_pLocationMapManager->BuildLocationMap();
+
+        ReviseInputImportCalls();
+
+        m_inOutCalls.clear(); // It will hold XX' output calls
+    }
+    else if (m_shaderStage == ShaderStageVertex)
+    {
+        ReassembleOutputExportCalls();
+
+        // For computing the shader hash
+        m_pContext->GetShaderResourceUsage(m_shaderStage)->inOutUsage.inOutLocMap =
+            m_pContext->GetShaderResourceUsage(ShaderStageFragment)->inOutUsage.inOutLocMap;
+    }
+    else
+    {
+        // TODO: Pack input/output in other stages is not supported
+        LLPC_NOT_IMPLEMENTED();
+    }
+}
+
+// =====================================================================================================================
+// Revise the location and element index fields of the fragment shaders input import functions
+void PatchResourceCollect::ReviseInputImportCalls()
+{
+    if (m_inOutCalls.empty())
+    {
+        return;
+    }
+
+    LLPC_ASSERT(m_shaderStage == ShaderStageFragment);
+
+    auto& inOutUsage = m_pContext->GetShaderResourceUsage(m_shaderStage)->inOutUsage;
+    auto& inputLocMap = inOutUsage.inputLocMap;
+    inputLocMap.clear();
+
+    IRBuilder<> builder(*m_pContext);
+
+    for (auto pCall : m_inOutCalls)
+    {
+        auto argCount = pCall->arg_size();
+        const bool isInterpolant = (argCount == 5);
+        uint32_t compIdx = 1;
+        uint32_t locOffset = 0;
+        if (isInterpolant)
+        {
+            compIdx = 2;
+            locOffset = cast<ConstantInt>(pCall->getOperand(1))->getZExtValue();
+        }
+
+        // Construct original InOutLocation from the location and elemIdx operands of the FS' input import call
+        InOutLocation origInLoc = {};
+        origInLoc.locationInfo.location = cast<ConstantInt>(pCall->getOperand(0))->getZExtValue() + locOffset;
+        origInLoc.locationInfo.component = cast<ConstantInt>(pCall->getOperand(compIdx))->getZExtValue();
+        origInLoc.locationInfo.half = false;
+
+        // Get the packed InOutLocation from locationMap
+        const InOutLocation* pNewInLoc = nullptr;
+        m_pLocationMapManager->FindMap(origInLoc, pNewInLoc);
+        LLPC_ASSERT(m_pLocationMapManager->FindMap(origInLoc, pNewInLoc));
+
+        // TODO: inputLocMap can be removed
+        inputLocMap[pNewInLoc->locationInfo.location] = InvalidValue;
+        inOutUsage.inOutLocMap[origInLoc.AsIndex()] = pNewInLoc->AsIndex();
+
+        // Re-write the input import call by using the new InOutLocation
+        SmallVector<Value*, 5> args;
+        std::string callName;
+        if (isInterpolant == false)
+        {
+            args.push_back(ConstantInt::get(m_pContext->Int32Ty(),
+                                            static_cast<uint32_t>(pNewInLoc->locationInfo.location)));
+            args.push_back(ConstantInt::get(m_pContext->Int32Ty(),
+                                        static_cast<uint32_t>(pNewInLoc->locationInfo.component)));
```

Comment:
You  can use builder.getInt32 for the constants instead. Also, I don't think the static_cast is necessary.

Reply:
Thanks. Fixed.